### PR TITLE
extmod/moduplatform: Improve implementation for PC ports.

### PR DESCRIPTION
**Pull request by stinos**
Fix identification of 32/64 bit and of the Windows platform and add a
platform string mimicking CPython for the latter.

### CODE REVIEW

File: extmod/moduplatform.c [hunk 1]
```diff
@@ -95,6 +107,10 @@
 #define MICROPY_HW_MCU_NAME     ""
 #endif
 
+#ifndef MICROPY_HAL_VERSION
+#define MICROPY_HAL_VERSION     ""
+#endif
+
 STATIC const MP_DEFINE_STR_OBJ(info_platform_obj, PLATFORM_SYSTEM "-" MICROPY_VERSION_STRING "-HAL" \
```
Comment (dpgeorge) [inline extmod/moduplatform.c:114]: I think we should remove the `"-HAL"` string here and make ports provide it (they could use SDK or IDF or whatever).  But that's definitely for a separate PR.

Comment (stinos) [reply extmod/moduplatform.c:114]: Yes. I first thought of adding logic to skip the -HAL part when MICROPY_HAL_VERSION is empty but would indeed be better to treat this as 'custom part of platform' like MICROPY_PLATFORM_POSTFIX and have that in mpconfig.h for consistency? Also MICROPY_HW_BOARD_NAME and MICROPY_HW_MCU_NAME are not used, these can be deleted?

**Comment by dpgeorge** [comment]
Merged in d42cba0d22cac812cc5a12f4670010b45932eafa

File: extmod/moduplatform.c (hunk 1 above)
Comment (dpgeorge) [reply extmod/moduplatform.c:114]: Yes, those latter two can be removed.

Feel free to open a PR to fix the above items.

**Pull request closed without merging**